src/main/java/com/sap/sgs/phosphor/fosstars/tool/github/MarkdownReporter.java

```diff
@@ -1,5 +1,184 @@
+      throws IOException {
+
+    Objects.requireNonNull(outputDirectory, "Oh no! Output directory is null!");
+    if (!Files.isDirectory(Paths.get(outputDirectory))) {
+      throw new FileNotFoundException(
+          String.format("Oh no! I could not find %s", outputDirectory));
+    }
+    this.outputDirectory = outputDirectory;
+    this.extraProjects = loadProjects(extraSourceFileName);
+  }
+
+  @Override
+  public void runFor(List<GitHubProject> projects) throws IOException {
+    List<GitHubProject> allProjects = merge(projects, extraProjects);
+    allProjects.sort(Comparator.comparing(project -> project.url().toString()));
+
+    String projectDetailsTemplate;
+    try (InputStream is = getClass().getResourceAsStream("MarkdownProjectDetailsTemplate.md")) {
+      projectDetailsTemplate = IOUtils.toString(is, "UTF-8");
+    }
+
+    StringBuilder sb = new StringBuilder();
+    for (GitHubProject project : allProjects) {
+      String projectPath = project.url().getPath().replaceFirst("/", "");
+
+      Path organizationDirectory = Paths.get(outputDirectory)
+          .resolve(project.organization().name());
+      if (!Files.isDirectory(organizationDirectory)) {
+        Files.createDirectories(organizationDirectory);
+      }
+
+      String details = projectDetailsTemplate
+          .replace("%PROJECT_NAME%", projectPath)
+          .replace("%DETAILS%", detailsOf(project));
+
+      String projectReportFilename = String.format("%s.md", project.name());
+      Files.write(
+          organizationDirectory.resolve(projectReportFilename),
+          details.getBytes());
+
+      String relativePathToDetails = String.format("%s/%s",
+          project.organization().name(), projectReportFilename);
+      String url = String.format("[%s](%s)", projectPath, project.url());
+      String label = String.format("[%s](%s)", labelOf(project), relativePathToDetails);
+      String line = PROJECT_LINE_TEMPLATE
+          .replace("%URL%", url)
+          .replace("%SCORE%", scoreOf(project))
+          .replace("%LABEL%", label)
+          .replace("%CONFIDENCE%", confidenceOf(project))
+          .replace("%DATE%", lastUpdateOf(project));
+      sb.append(line).append("\n");
+    }
+
+    try (InputStream is = getClass().getResourceAsStream("MarkdownReporterMainTemplate.md")) {
+      String template = IOUtils.toString(is, "UTF-8");
+      String content = template.replace("%PROJECT_TABLE%", sb.toString());
+
+      Path path = Paths.get(outputDirectory).resolve(REPORT_FILENAME);
+      System.out.printf("[+] Storing a report to %s%n", path);
+      Files.write(path, content.getBytes());
+    }
+  }
+
+  /**
+   * Prepares a description how a rating was calculated for a project.
+   *
+   * @param project The project.
+   * @return The details of the rating calculation.
+   */
+  private static String detailsOf(GitHubProject project) {
+    if (!project.ratingValue().isPresent()) {
+      return "unknown";
```

Comment:
looks like a static final variable

Reply:
I can do that. Most likely the report is going to be made nicer, so this `static final` will likely be removed.